diff -Naur ../official/public/v1/js/ff/accounts/reconcile.js ./public/v1/js/ff/accounts/reconcile.js
--- ../official/public/v1/js/ff/accounts/reconcile.js	2022-05-06 16:20:55.829339578 +0200
+++ ./public/v1/js/ff/accounts/reconcile.js	2022-07-16 08:22:54.605171977 +0200
@@ -21,6 +21,7 @@
 var balanceDifference = 0;
 var difference = 0;
 var selectedAmount = 0;
+var initialSelectedAmount = 0;
 var reconcileStarted = false;
 var changedBalances = false;
 
@@ -41,6 +42,9 @@
             difference = balanceDifference - selectedAmount;
             updateDifference();
 
+            trigger_smart_reconcile();
+            difference = balanceDifference - selectedAmount;
+            updateDifference();
         }
         changedBalances = true;
     });
@@ -69,6 +73,98 @@
 
 });
 
+function smart_reconcile(amounts, target_sum) {
+    let d_size = amounts.length;
+    let max_sum = amounts.reduce((a, b) => a + b);
+    // Define number of maximum items to uncheck in order to keep short computing time
+    // Here are some pre-computed values ensuring less than 1e7 combinations to explore
+    // 10 -> 10, 20 -> 20, 22 -> 22, 24 -> 12, 26 -> 9, 30 -> 8, 40 -> 6, 50 -> 5, 100 -> 4
+    // next line is a magic formula fitting these values
+    let max_nb_items = (d_size <= 22) ? d_size : Math.floor( 13*(d_size-22)**-0.3 );
+    console.log('debug smart_reconcile:', target_sum, max_sum, max_nb_items);
+    // Result is list of items NOT to check
+    let result = smart_reconcile_aux(amounts, 0, max_nb_items, 0, max_sum-target_sum);
+    return result;
+}
+
+// Recursive function, try to add the amounts after the 'nth' one to achieve 'target_sum'
+// without exceeding maximum number of items 'remain_to_take'
+// Returns null if not possible else boolean array listing which items to check
+function smart_reconcile_aux(amounts, nth_number, remain_to_take, current_sum, target_sum) {
+    // Stop if no more transactions can be checked
+    if (Math.abs(current_sum - target_sum) < 0.01) return Array(amounts.length-nth_number).fill(false);
+    if (remain_to_take == 0 || nth_number == amounts.length) return null;
+
+    // Try without using current number
+    let result = smart_reconcile_aux(amounts, nth_number+1, remain_to_take, current_sum, target_sum);
+    if (result) {
+        result.unshift(false);
+        return result;
+    }
+
+    // Try using current number
+    let new_amount = amounts[nth_number];
+    let new_sum = current_sum + new_amount;
+    if (Math.abs(new_sum - target_sum) < 1e-8) return [true].concat(Array(amounts.length-nth_number-1).fill(false));
+    result = smart_reconcile_aux(amounts, nth_number+1, remain_to_take-1, new_sum, target_sum);
+    if (result) result.unshift(true);
+    return result;
+}
+
+function trigger_smart_reconcile() {
+    // Gather all data
+    let amounts = [];
+    $('.reconcile_checkbox').each(function (i, v) {
+        var check = $(v);
+        amounts.push(parseFloat(check.val()));
+    });
+
+    // Call core function, better to serch for items to uncheck than the ones to check
+    var items_to_uncheck = smart_reconcile(amounts, initialSelectedAmount - balanceDifference);
+    if (items_to_uncheck) {
+        // Set checkboxes
+        $('.reconcile_checkbox').each(function (i, v) {
+            var check = $(v);
+            changeReconciledBoxAndUpdate(check, !items_to_uncheck[i]);
+        });
+    } else {
+        console.log('Cant find a solution for smart reconcile');
+    }
+}
+
+// Change status of el to targetState and update all variables
+// If targetState is null, then assumes that state is alreay changed and only update variables
+function changeReconciledBoxAndUpdate(el, targetState) {
+    var amount = parseFloat(el.val());
+    var journalId = parseInt(el.data('id'));
+    var identifier = 'checked_' + journalId;
+    console.log('in changeReconciledBoxAndUpdate(' + journalId + ') with amount ' + amount + ' selected amount ' + selectedAmount + ' and targetState=' + targetState);
+
+    if (targetState != null) {
+        // do nothing if line is already in target state
+        if (el.prop('checked') === targetState) {
+            return;
+        }
+        el.prop('checked', targetState);
+    }
+
+    // if checked, add to selected amount
+    if (el.prop('checked') === true && el.data('younger') === false) {
+        selectedAmount = selectedAmount - amount;
+        //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
+        localStorage.setItem(identifier, 'true');
+    }
+    // if unchecked, substract from selected amount
+    if (el.prop('checked') === false && el.data('younger') === false) {
+        selectedAmount = selectedAmount + amount;
+        //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
+        localStorage.setItem(identifier, 'false');
+    }
+    difference = balanceDifference - selectedAmount;
+    //console.log('Difference is now ' + difference);
+    updateDifference();
+}
+
 function selectAllReconcile(e) {
     // loop all, check.
     var el = $(e.target);
@@ -83,29 +179,7 @@
 
     $('.reconcile_checkbox').each(function (i, v) {
         var check = $(v);
-        var amount = parseFloat(check.val());
-        var journalId = parseInt(check.data('id'));
-        var identifier = 'checked_' + journalId;
-        console.log('in selectAllReconcile(' + journalId + ') with amount ' + amount + ' and selected amount ' + selectedAmount);
-
-        // do nothing if line is already in target state
-        if (check.prop('checked') === doCheck )
-            return;
-    
-        check.prop('checked', doCheck);
-        // if checked, add to selected amount
-        if (doCheck === true && check.data('younger') === false) {
-            selectedAmount = selectedAmount - amount;
-            //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
-            localStorage.setItem(identifier, 'true');
-        }
-        if (doCheck === false && check.data('younger') === false) {
-            selectedAmount = selectedAmount + amount;
-            //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
-            localStorage.setItem(identifier, 'false');
-        }
-        difference = balanceDifference - selectedAmount;
-        //console.log('Difference is now ' + difference);
+        changeReconciledBoxAndUpdate(check, doCheck);
     });
 
     updateDifference();
@@ -153,26 +227,7 @@
  * @param e
  */
 function checkReconciledBox(e) {
-
-    var el = $(e.target);
-    var amount = parseFloat(el.val());
-    var journalId = parseInt(el.data('id'));
-    var identifier = 'checked_' + journalId;
-    //console.log('in checkReconciledBox(' + journalId + ') with amount ' + amount + ' and selected amount ' + selectedAmount);
-    // if checked, add to selected amount
-    if (el.prop('checked') === true && el.data('younger') === false) {
-        selectedAmount = selectedAmount - amount;
-        //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
-        localStorage.setItem(identifier, 'true');
-    }
-    if (el.prop('checked') === false && el.data('younger') === false) {
-        selectedAmount = selectedAmount + amount;
-        //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
-        localStorage.setItem(identifier, 'false');
-    }
-    difference = balanceDifference - selectedAmount;
-    //console.log('Difference is now ' + difference);
-    updateDifference();
+    changeReconciledBoxAndUpdate($(e.target), null);
 }
 
 
@@ -266,6 +321,7 @@
 
     difference = balanceDifference - selectedAmount;
     updateDifference();
+    initialSelectedAmount = selectedAmount;
 
     // loop al placed checkboxes and check them if necessary.
     restoreFromLocalStorage();
@@ -331,12 +387,12 @@
 }
 
 function updateDifference() {
-    console.log('in updateDifference()');
+    //console.log('in updateDifference()');
     var addClass = 'text-info';
-    if (difference > 0) {
+    if (difference > 0.01) {
         addClass = 'text-success';
     }
-    if (difference < 0) {
+    if (difference < -0.01) {
         addClass = 'text-danger';
     }
     $('#difference').addClass(addClass).text(accounting.formatMoney(difference));
diff -Naur ../official/resources/assets/js/components/transactions/CreateTransaction.vue ./resources/assets/js/components/transactions/CreateTransaction.vue
--- ../official/resources/assets/js/components/transactions/CreateTransaction.vue	2022-01-30 07:42:09.276120228 +0100
+++ ./resources/assets/js/components/transactions/CreateTransaction.vue	2022-07-16 08:07:47.674616594 +0200
@@ -134,35 +134,14 @@
                       :transactionType="transactionType"
                       v-bind:title="$t('form.foreign_amount')"
                   ></foreign-amount>
-                </div>
-                <div id="optional-info" class="col-lg-4">
                   <budget
                       v-model="transaction.budget"
                       :error="transaction.errors.budget_id"
                       :no_budget="$t('firefly.none_in_select_list')"
                       :transactionType="transactionType"
                   ></budget>
-                  <category
-                      v-model="transaction.category"
-                      :error="transaction.errors.category"
-                      :transactionType="transactionType"
-                  ></category>
-                  <piggy-bank
-                      v-model="transaction.piggy_bank"
-                      :error="transaction.errors.piggy_bank"
-                      :no_piggy_bank="$t('firefly.no_piggy_bank')"
-                      :transactionType="transactionType"
-                  ></piggy-bank>
-                  <tags
-                      v-model="transaction.tags"
-                      :error="transaction.errors.tags"
-                  ></tags>
-                  <bill
-                      v-model="transaction.bill"
-                      :error="transaction.errors.bill_id"
-                      :no_bill="$t('firefly.none_in_select_list')"
-                      :transactionType="transactionType"
-                  ></bill>
+                </div>
+                <div id="optional-info" class="col-lg-4">
                   <custom-transaction-fields
                       v-model="transaction.custom_fields"
                       :error="transaction.errors.custom_errors"
diff -Naur ../official/resources/views/accounts/reconcile/index.twig ./resources/views/accounts/reconcile/index.twig
--- ../official/resources/views/accounts/reconcile/index.twig	2022-05-06 16:20:55.873340118 +0200
+++ ./resources/views/accounts/reconcile/index.twig	2022-07-16 08:07:47.678616704 +0200
@@ -134,4 +134,7 @@
         var selectRangeAndBalance = '{{ 'select_range_and_balance'|_|escape('js') }}';
     </script>
     <script src="v1/js/ff/accounts/reconcile.js?v={{ FF_VERSION }}" type="text/javascript" nonce="{{ JS_NONCE }}"></script>
+    <script type="text/javascript" nonce="{{ JS_NONCE }}">
+        document.addEventListener('DOMContentLoaded', startReconcile, false);
+    </script>
 {% endblock %}
diff -Naur ../official/resources/views/index.twig ./resources/views/index.twig
--- ../official/resources/views/index.twig	2022-05-06 16:20:55.873340118 +0200
+++ ./resources/views/index.twig	2022-07-16 08:07:47.678616704 +0200
@@ -4,7 +4,6 @@
     {{ Breadcrumbs.render }}
 {% endblock %}
 {% block content %}
-    {% include 'partials.boxes' %}
     <div class="row">
         <div class="col-lg-8 col-md-12 col-sm-12">
             {# ACCOUNTS #}
@@ -36,22 +35,6 @@
                     </a>
                 </div>
             </div>
-            {# CATEGORIES #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('categories.index') }}" title="{{ 'categories'|_ }}">{{ 'categories'|_ }}</a></h3>
-
-                </div>
-                <div class="box-body">
-                    <canvas id="categories-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('categories.index') }}" class="btn btn-default button-sm">
-                        <span class="fa fa-bookmark"></span>
-                        <span>{{ 'go_to_categories'|_ }}</span>
-                    </a>
-                </div>
-            </div>
         </div>
         <div class="col-lg-4 col-md-6 col-sm-12">
 
@@ -106,64 +89,6 @@
                     </div>
                 {% endfor %}
             </div>
-
-            {% if billCount > 0 %}
-                {# BILLS #}
-                <div class="box">
-                    <div class="box-header with-border">
-                        <h3 class="box-title"><a href="{{ route('bills.index') }}" title="{{ 'bills'|_ }}">{{ 'bills'|_ }}</a></h3>
-
-                    </div>
-                    <div class="box-body">
-                        <div style="width:100%;margin:0 auto;">
-                            <canvas id="bills-chart" style="width:100%;height:200px;" height="200"></canvas>
-                        </div>
-                    </div>
-                    <div class="box-footer">
-                        <a href="{{ route('bills.index') }}" class="btn btn-default button-sm"><span
-                                    class="fa fa-calendar"></span> {{ 'go_to_bills'|_ }}</a>
-                    </div>
-                </div>
-            {% endif %}
-
-            {# box for piggy bank data (JSON) #}
-            <div id="piggy_bank_overview">
-
-            </div>
-        </div>
-    </div>
-    <div class="row">
-        <div class="col-lg-12 col-md-12 col-sm-12">
-            {# EXPENSE ACCOUNTS #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('accounts.index',['expense']) }}"
-                                             title="{{ 'expense_accounts'|_ }}">{{ 'expense_accounts'|_ }}</a>
-                    </h3>
-                </div>
-                <div class="box-body">
-                    <canvas id="expense-accounts-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('accounts.index', ['expense']) }}" class="btn btn-default button-sm"><span
-                                class="fa fa-shopping-cart"></span> {{ 'go_to_expense_accounts'|_ }}</a>
-                </div>
-            </div>
-            {# OPTIONAL REVENUE ACCOUNTS #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('accounts.index',['revenue']) }}"
-                                             title="{{ 'revenue_accounts'|_ }}">{{ 'revenue_accounts'|_ }}</a></h3>
-
-                </div>
-                <div class="box-body">
-                    <canvas id="revenue-accounts-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('accounts.index', ['revenue']) }}" class="btn btn-default button-sm"><span
-                                class="fa fa-download"></span> {{ 'go_to_revenue_accounts'|_ }}</a>
-                </div>
-            </div>
         </div>
     </div>
 {% endblock %}
diff -Naur ../official/resources/views/layout/default.twig ./resources/views/layout/default.twig
--- ../official/resources/views/layout/default.twig	2022-05-06 16:20:55.873340118 +0200
+++ ./resources/views/layout/default.twig	2022-07-16 08:07:47.678616704 +0200
@@ -84,11 +84,22 @@
                         </span>
                     </li>
 
-                    <li class="dropdown user user-menu">
-                    <span style="cursor:default;color:#fff;padding: 15px;display: block;line-height: 20px;">
-                        <span class="hidden-xs">{{ Auth.user.email }}</span>
-                    </span>
+                    <li>
+                        <a href="{{ route('transactions.create', 'withdrawal') }}">
+                            <span class="fa fa-long-arrow-left bg-red" style="font-size:20px;border-radius:50%"></span>
+                        </a>
                     </li>
+                    <li>
+                        <a href="{{ route('transactions.create', 'deposit') }}">
+                            <span class="fa fa-long-arrow-right bg-green" style="font-size:20px;border-radius:50%"></span>
+                        </a>
+                    </li>
+                    <li>
+                        <a href="{{ route('transactions.create', 'transfer') }}">
+                            <span class="fa fa-exchange bg-blue" style="font-size:20px;border-radius:50%"></span>
+                        </a>
+                    </li>
+
                     <li id="sidebar-toggle">
                         <a href="#" data-toggle="control-sidebar"><span class="fa fa-plus-circle"></span></a>
                     </li>
diff -Naur ../official/resources/views/list/groups-tiny.twig ./resources/views/list/groups-tiny.twig
--- ../official/resources/views/list/groups-tiny.twig	2022-01-30 07:42:09.304120315 +0100
+++ ./resources/views/list/groups-tiny.twig	2022-07-16 08:07:47.682616812 +0200
@@ -7,7 +7,12 @@
             {{ transaction.description }}
 
             <span class="pull-right small">
-          {% if transaction.transaction_type_type == 'Deposit' %}
+          {% if transaction.date > date('now') %}
+              {#  future transactions:  #}
+              <span class="text-secondary">
+                  {{ formatAmountBySymbol(transaction.amount|abs, transaction.currency_symbol, transaction.currency_decimal_places, false) }}
+              </span>
+          {% elseif transaction.transaction_type_type == 'Deposit' %}
               {{ formatAmountBySymbol(transaction.amount*-1, transaction.currency_symbol, transaction.currency_decimal_places) }}
               {% if null != transaction.foreign_amount %}
                   ({{ formatAmountBySymbol(transaction.foreign_amount*-1, transaction.foreign_currency_symbol, transaction.foreign_currency_decimal_places) }})
diff -Naur ../official/app/Http/Controllers/Account/ReconcileController.php ./app/Http/Controllers/Account/ReconcileController.php
--- ../official/app/Http/Controllers/Account/ReconcileController.php	2022-05-06 16:20:55.797339186 +0200
+++ ./app/Http/Controllers/Account/ReconcileController.php	2022-07-16 08:04:19.161365342 +0200
@@ -105,19 +105,15 @@
         }
         $currency = $this->accountRepos->getAccountCurrency($account) ?? app('amount')->getDefaultCurrency();
 
-        // no start or end:
-        $range = app('preferences')->get('viewRange', '1M')->data;
-
-        // get start and end
-
+        // compute values of 'start' and 'end' whatever the preferred range is
         if (null === $start && null === $end) {
-
-            /** @var Carbon $start */
-            $start = clone session('start', app('navigation')->startOfPeriod(new Carbon, $range));
-            /** @var Carbon $end */
-            $end = clone session('end', app('navigation')->endOfPeriod(new Carbon, $range));
+           /** @var Carbon $start */
+           $start = clone (new Carbon)->subMonth();
+           /** @var Carbon $end */
+           $end = clone (new Carbon);
         }
         if (null === $end) {
+            $range = app('preferences')->get('viewRange', '1M')->data;
             /** @var Carbon $end */
             $end = app('navigation')->endOfPeriod($start, $range);
         }
diff -Naur ../official/app/Jobs/CreateRecurringTransactions.php ./app/Jobs/CreateRecurringTransactions.php
--- ../official/app/Jobs/CreateRecurringTransactions.php	2022-05-06 16:20:55.801339234 +0200
+++ ./app/Jobs/CreateRecurringTransactions.php	2022-07-16 08:04:19.165365429 +0200
@@ -87,6 +87,8 @@
         $this->submitted         = 0;
         $this->executed          = 0;
         $this->created           = 0;
+        $this->days_max_late     = 3;
+        $this->days_max_early    = 8;
 
         Log::debug(sprintf('Created new CreateRecurringTransactions("%s")', $this->date->format('Y-m-d')));
     }
@@ -309,10 +311,10 @@
             );
 
             // start looping from $startDate to today perhaps we have a hit?
-            // add two days to $this->date so we always include the weekend.
-            $includeWeekend = clone $this->date;
-            $includeWeekend->addDays(2);
-            $occurrences = $this->repository->getOccurrencesInRange($repetition, $recurrence->first_date, $includeWeekend);
+            // add few days to anticipate next occurrences
+            $endDate = clone $this->date;
+            $endDate->addDays($this->days_max_early);
+            $occurrences = $this->repository->getOccurrencesInRange($repetition, $recurrence->first_date, $endDate);
 
             unset($includeWeekend);
 
@@ -360,11 +362,18 @@
     private function handleOccurrence(Recurrence $recurrence, RecurrenceRepetition $repetition, Carbon $date): ?TransactionGroup
     {
         $date->startOfDay();
-        if ($date->ne($this->date)) {
 
+        // allow some margin in the future (to anticipate) and in the past (in case cron was not executed)
+        $datediff = $date->diffInDays($this->date);
+        if ($datediff > $this->days_max_early) {
+            //print "date too much in the future\n";
             return null;
         }
-        Log::debug(sprintf('%s IS today (%s)', $date->format('Y-m-d'), $this->date->format('Y-m-d')));
+        if ($datediff < -$this->days_max_late) {
+            //print "date too much in the past\n";
+            return null;
+        }
+        Log::debug(sprintf('%s IS close to today (%s)', $date->format('Y-m-d'), $this->date->format('Y-m-d')));
 
         // count created journals on THIS day.
         $journalCount = $this->repository->getJournalCount($recurrence, $date, $date);
diff -Naur ../official/app/Support/Http/Controllers/GetConfigurationData.php ./app/Support/Http/Controllers/GetConfigurationData.php
--- ../official/app/Support/Http/Controllers/GetConfigurationData.php	2022-05-06 16:20:55.813339381 +0200
+++ ./app/Support/Http/Controllers/GetConfigurationData.php	2022-07-16 08:07:47.662616268 +0200
@@ -52,7 +52,7 @@
             E_COMPILE_ERROR | E_RECOVERABLE_ERROR | E_ERROR | E_CORE_ERROR => 'E_COMPILE_ERROR|E_RECOVERABLE_ERROR|E_ERROR|E_CORE_ERROR',
         ];
 
-        return $array[$value] ?? (string) $value;
+        return $array[$value] ?? (string)$value;
     }
 
     /**
@@ -72,7 +72,7 @@
                 $currentStep = $options;
 
                 // get the text:
-                $currentStep['intro'] = (string) trans('intro.' . $route . '_' . $key);
+                $currentStep['intro'] = (string)trans('intro.' . $route . '_' . $key);
 
                 // save in array:
                 $steps[] = $currentStep;
@@ -93,7 +93,7 @@
      */
     protected function getDateRangeConfig(): array // get configuration + get preferences.
     {
-        $viewRange = (string) app('preferences')->get('viewRange', '1M')->data;
+        $viewRange = (string)app('preferences')->get('viewRange', '1M')->data;
         /** @var Carbon $start */
         $start = session('start');
         /** @var Carbon $end */
@@ -134,31 +134,32 @@
         /** @var Carbon $todayEnd */
         $todayEnd = app('navigation')->endOfPeriod($todayStart, $viewRange);
         if ($todayStart->ne($start) || $todayEnd->ne($end)) {
-            $ranges[ucfirst((string) trans('firefly.today'))] = [$todayStart, $todayEnd];
+            $ranges[ucfirst((string)trans('firefly.today'))] = [$todayStart, $todayEnd];
         }
 
         // last seven days:
-        $seven          = Carbon::now()->subDays(7);
-        $index          = (string) trans('firefly.last_seven_days');
-        $ranges[$index] = [$seven, new Carbon];
+        $custom_begin   = Carbon::now()->subDays(14);
+        $custom_end     = Carbon::now()->addDays(7);
+        $index          = (string)trans('firefly.last_seven_days');
+        $ranges[$index] = [$custom_begin, $custom_end];
 
         // last 30 days:
         $thirty         = Carbon::now()->subDays(30);
-        $index          = (string) trans('firefly.last_thirty_days');
+        $index          = (string)trans('firefly.last_thirty_days');
         $ranges[$index] = [$thirty, new Carbon];
 
         // everything
-        $index          = (string) trans('firefly.everything');
+        $index          = (string)trans('firefly.everything');
         $ranges[$index] = [$first, new Carbon];
 
         return [
             'title'         => $title,
             'configuration' => [
-                'apply'       => (string) trans('firefly.apply'),
-                'cancel'      => (string) trans('firefly.cancel'),
-                'from'        => (string) trans('firefly.from'),
-                'to'          => (string) trans('firefly.to'),
-                'customRange' => (string) trans('firefly.customRange'),
+                'apply'       => (string)trans('firefly.apply'),
+                'cancel'      => (string)trans('firefly.cancel'),
+                'from'        => (string)trans('firefly.from'),
+                'to'          => (string)trans('firefly.to'),
+                'customRange' => (string)trans('firefly.customRange'),
                 'start'       => $start->format('Y-m-d'),
                 'end'         => $end->format('Y-m-d'),
                 'ranges'      => $ranges,
@@ -189,7 +190,7 @@
                     $currentStep = $options;
 
                     // get the text:
-                    $currentStep['intro'] = (string) trans('intro.' . $route . '_' . $specificPage . '_' . $key);
+                    $currentStep['intro'] = (string)trans('intro.' . $route . '_' . $specificPage . '_' . $key);
 
                     // save in array:
                     $steps[] = $currentStep;
@@ -207,7 +208,7 @@
     protected function verifyRecurringCronJob(): void
     {
         $config   = app('fireflyconfig')->get('last_rt_job', 0);
-        $lastTime = (int) $config->data;
+        $lastTime = (int)$config->data;
         $now      = time();
         Log::debug(sprintf('verifyRecurringCronJob: last time is %d ("%s"), now is %d', $lastTime, $config->data, $now));
         if (0 === $lastTime) {
