diff -Naur ../official/public/v1/js/ff/accounts/reconcile.js ./public/v1/js/ff/accounts/reconcile.js
--- ../official/public/v1/js/ff/accounts/reconcile.js	2022-05-06 16:20:55.829339578 +0200
+++ ./public/v1/js/ff/accounts/reconcile.js	2023-05-28 06:05:32.582831424 +0200
@@ -21,6 +21,7 @@
 var balanceDifference = 0;
 var difference = 0;
 var selectedAmount = 0;
+var initialSelectedAmount = 0;
 var reconcileStarted = false;
 var changedBalances = false;
 
@@ -41,6 +42,9 @@
             difference = balanceDifference - selectedAmount;
             updateDifference();
 
+            trigger_smart_reconcile();
+            difference = balanceDifference - selectedAmount;
+            updateDifference();
         }
         changedBalances = true;
     });
@@ -69,6 +73,107 @@
 
 });
 
+function smart_reconcile(amounts, target_sum) {
+    let d_size = amounts.length;
+    let max_sum = amounts.reduce((a, b) => a + b);
+    // Define number of maximum items to exclude in order to keep short computing time
+    // Here are some pre-computed values ensuring less than 1e7 combinations to explore
+    // 10 -> 10, 20 -> 20, 22 -> 22, 24 -> 12, 26 -> 9, 30 -> 8, 40 -> 6, 50 -> 5, 100 -> 4
+    // next line is a magic formula fitting these values
+    let max_nb_unchecked_items = (d_size <= 22) ? d_size : Math.floor( 13*(d_size-22)**-0.3 );
+    console.log('debug smart_reconcile:', target_sum, max_sum, max_nb_unchecked_items);
+    // Result will be list of items to EXCLUDE from reconcile
+    let result = solve_knapsack(amounts, amounts.length, max_nb_unchecked_items, 0, max_sum-target_sum);
+    return result;
+}
+
+// Find combination of transactions whose sum equals to target, with a preference
+// to pick minimum number of transactions and new transactions first. (remember
+// that pick = exclude some reconcile).
+// Tree exploration through recursive function:
+// Add transactions from index-1 to 0 to achieve target_sum without picking more
+// than remain_to_take items 
+// Returns null if not possible, else boolean array listing which transactions to pick
+function solve_knapsack(amounts, index, remain_to_take, current_sum, target_sum) {
+    // Stop if no more transactions can be checked
+    if (Math.abs(current_sum - target_sum) < 0.001) return Array(index).fill(false);
+    if (remain_to_take == 0 || index == 0) return null;
+
+    // Try not to pick current transaction
+    let result = solve_knapsack(amounts, index-1, remain_to_take, current_sum, target_sum);
+    if (result) {
+        result.push(false);
+        return result;
+    }
+
+    // Try picking current transaction
+    let new_amount = amounts[index-1];
+    let new_sum = current_sum + new_amount;
+    if (Math.abs(new_sum - target_sum) < 0.001) return Array(index-1).fill(false).concat([true]);
+    result = solve_knapsack(amounts, index-1, remain_to_take-1, new_sum, target_sum);
+    if (result) {
+        result.push(true);
+        return result;
+    }
+
+    // No solution
+    return null;
+}
+
+function trigger_smart_reconcile() {
+    // Gather all data
+    let amounts = [];
+    $('.reconcile_checkbox').each(function (i, v) {
+        var check = $(v);
+        amounts.push(parseFloat(check.val()));
+    });
+
+    // Call core function, better to serch for items to uncheck than the ones to check
+    var items_to_uncheck = smart_reconcile(amounts, initialSelectedAmount - balanceDifference);
+    if (items_to_uncheck) {
+        // Set checkboxes
+        $('.reconcile_checkbox').each(function (i, v) {
+            var check = $(v);
+            changeReconciledBoxAndUpdate(check, !items_to_uncheck[i]);
+        });
+    } else {
+        console.log('Cant find a solution for smart reconcile');
+    }
+}
+
+// Change status of el to targetState and update all variables
+// If targetState is null, then assumes that state is alreay changed and only update variables
+function changeReconciledBoxAndUpdate(el, targetState) {
+    var amount = parseFloat(el.val());
+    var journalId = parseInt(el.data('id'));
+    var identifier = 'checked_' + journalId;
+    console.log('in changeReconciledBoxAndUpdate(' + journalId + ') with amount ' + amount + ' selected amount ' + selectedAmount + ' and targetState=' + targetState);
+
+    if (targetState != null) {
+        // do nothing if line is already in target state
+        if (el.prop('checked') === targetState) {
+            return;
+        }
+        el.prop('checked', targetState);
+    }
+
+    // if checked, add to selected amount
+    if (el.prop('checked') === true && el.data('younger') === false) {
+        selectedAmount = selectedAmount - amount;
+        //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
+        localStorage.setItem(identifier, 'true');
+    }
+    // if unchecked, substract from selected amount
+    if (el.prop('checked') === false && el.data('younger') === false) {
+        selectedAmount = selectedAmount + amount;
+        //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
+        localStorage.setItem(identifier, 'false');
+    }
+    difference = balanceDifference - selectedAmount;
+    //console.log('Difference is now ' + difference);
+    updateDifference();
+}
+
 function selectAllReconcile(e) {
     // loop all, check.
     var el = $(e.target);
@@ -83,29 +188,7 @@
 
     $('.reconcile_checkbox').each(function (i, v) {
         var check = $(v);
-        var amount = parseFloat(check.val());
-        var journalId = parseInt(check.data('id'));
-        var identifier = 'checked_' + journalId;
-        console.log('in selectAllReconcile(' + journalId + ') with amount ' + amount + ' and selected amount ' + selectedAmount);
-
-        // do nothing if line is already in target state
-        if (check.prop('checked') === doCheck )
-            return;
-    
-        check.prop('checked', doCheck);
-        // if checked, add to selected amount
-        if (doCheck === true && check.data('younger') === false) {
-            selectedAmount = selectedAmount - amount;
-            //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
-            localStorage.setItem(identifier, 'true');
-        }
-        if (doCheck === false && check.data('younger') === false) {
-            selectedAmount = selectedAmount + amount;
-            //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
-            localStorage.setItem(identifier, 'false');
-        }
-        difference = balanceDifference - selectedAmount;
-        //console.log('Difference is now ' + difference);
+        changeReconciledBoxAndUpdate(check, doCheck);
     });
 
     updateDifference();
@@ -153,26 +236,7 @@
  * @param e
  */
 function checkReconciledBox(e) {
-
-    var el = $(e.target);
-    var amount = parseFloat(el.val());
-    var journalId = parseInt(el.data('id'));
-    var identifier = 'checked_' + journalId;
-    //console.log('in checkReconciledBox(' + journalId + ') with amount ' + amount + ' and selected amount ' + selectedAmount);
-    // if checked, add to selected amount
-    if (el.prop('checked') === true && el.data('younger') === false) {
-        selectedAmount = selectedAmount - amount;
-        //console.log('checked = true and younger = false so selected amount = ' + selectedAmount);
-        localStorage.setItem(identifier, 'true');
-    }
-    if (el.prop('checked') === false && el.data('younger') === false) {
-        selectedAmount = selectedAmount + amount;
-        //console.log('checked = false and younger = false so selected amount = ' + selectedAmount);
-        localStorage.setItem(identifier, 'false');
-    }
-    difference = balanceDifference - selectedAmount;
-    //console.log('Difference is now ' + difference);
-    updateDifference();
+    changeReconciledBoxAndUpdate($(e.target), null);
 }
 
 
@@ -266,6 +330,7 @@
 
     difference = balanceDifference - selectedAmount;
     updateDifference();
+    initialSelectedAmount = selectedAmount;
 
     // loop al placed checkboxes and check them if necessary.
     restoreFromLocalStorage();
@@ -331,12 +396,12 @@
 }
 
 function updateDifference() {
-    console.log('in updateDifference()');
+    //console.log('in updateDifference()');
     var addClass = 'text-info';
-    if (difference > 0) {
+    if (difference > 0.01) {
         addClass = 'text-success';
     }
-    if (difference < 0) {
+    if (difference < -0.01) {
         addClass = 'text-danger';
     }
     $('#difference').addClass(addClass).text(accounting.formatMoney(difference));
diff -Naur ../official/resources/assets/js/components/transactions/CreateTransaction.vue ./resources/assets/js/components/transactions/CreateTransaction.vue
--- ../official/resources/assets/js/components/transactions/CreateTransaction.vue	2023-03-15 08:36:59.151380724 +0100
+++ ./resources/assets/js/components/transactions/CreateTransaction.vue	2023-03-15 08:37:07.299487107 +0100
@@ -134,35 +134,14 @@
                       :transactionType="transactionType"
                       v-bind:title="$t('form.foreign_amount')"
                   ></foreign-amount>
-                </div>
-                <div id="optional-info" class="col-lg-4">
                   <budget
                       v-model="transaction.budget"
                       :error="transaction.errors.budget_id"
                       :no_budget="$t('firefly.none_in_select_list')"
                       :transactionType="transactionType"
                   ></budget>
-                  <category
-                      v-model="transaction.category"
-                      :error="transaction.errors.category"
-                      :transactionType="transactionType"
-                  ></category>
-                  <piggy-bank
-                      v-model="transaction.piggy_bank"
-                      :error="transaction.errors.piggy_bank"
-                      :no_piggy_bank="$t('firefly.no_piggy_bank')"
-                      :transactionType="transactionType"
-                  ></piggy-bank>
-                  <tags
-                      v-model="transaction.tags"
-                      :error="transaction.errors.tags"
-                  ></tags>
-                  <bill
-                      v-model="transaction.bill"
-                      :error="transaction.errors.bill_id"
-                      :no_bill="$t('firefly.none_in_select_list')"
-                      :transactionType="transactionType"
-                  ></bill>
+                </div>
+                <div id="optional-info" class="col-lg-4">
                   <custom-transaction-fields
                       v-model="transaction.custom_fields"
                       :error="transaction.errors.custom_errors"
diff -Naur ../official/resources/lang/en_US/firefly.php ./resources/lang/en_US/firefly.php
--- ../official/resources/lang/en_US/firefly.php	2023-07-01 10:55:37.334922424 +0200
+++ ./resources/lang/en_US/firefly.php	2023-07-01 10:55:39.199023096 +0200
@@ -1274,6 +1274,7 @@
     'pref_view_range_help'                      => 'Some charts are automatically grouped in periods. Your budgets will also be grouped in periods. What period would you prefer?',
     'pref_1D'                                   => 'One day',
     'pref_1W'                                   => 'One week',
+    'pref_3W'                                   => 'Three weeks centered',
     'pref_1M'                                   => 'One month',
     'pref_3M'                                   => 'Three months (quarter)',
     'pref_6M'                                   => 'Six months',
diff -Naur ../official/resources/lang/fr_FR/firefly.php ./resources/lang/fr_FR/firefly.php
--- ../official/resources/lang/fr_FR/firefly.php	2023-07-01 10:55:37.338922651 +0200
+++ ./resources/lang/fr_FR/firefly.php	2023-07-01 10:55:39.203023295 +0200
@@ -1314,6 +1314,7 @@
     'pref_view_range_help'                      => 'Certains graphiques sont automatiquement regroupés par périodes. Vos budgets seront aussi organisés par périodes. Quelle période préférez-vous ?',
     'pref_1D'                                   => 'Un jour',
     'pref_1W'                                   => 'Une semaine',
+    'pref_3W'                                   => 'Trois semaines centrees',
     'pref_1M'                                   => 'Un mois',
     'pref_3M'                                   => 'Trois mois (trimestre)',
     'pref_6M'                                   => 'Six mois',
diff -Naur ../official/resources/views/accounts/reconcile/index.twig ./resources/views/accounts/reconcile/index.twig
--- ../official/resources/views/accounts/reconcile/index.twig	2022-12-29 16:01:27.132300233 +0100
+++ ./resources/views/accounts/reconcile/index.twig	2022-12-29 16:01:29.804338669 +0100
@@ -134,4 +134,7 @@
         var selectRangeAndBalance = '{{ 'select_range_and_balance'|_|escape('js') }}';
     </script>
     <script src="v1/js/ff/accounts/reconcile.js?v={{ FF_VERSION }}" type="text/javascript" nonce="{{ JS_NONCE }}"></script>
+    <script type="text/javascript" nonce="{{ JS_NONCE }}">
+        document.addEventListener('DOMContentLoaded', startReconcile, false);
+    </script>
 {% endblock %}
diff -Naur ../official/resources/views/index.twig ./resources/views/index.twig
--- ../official/resources/views/index.twig	2023-03-15 08:36:59.187381194 +0100
+++ ./resources/views/index.twig	2023-03-15 08:37:07.335487577 +0100
@@ -4,7 +4,6 @@
     {{ Breadcrumbs.render }}
 {% endblock %}
 {% block content %}
-    {% include 'partials.boxes' %}
     <div class="row">
         <div class="col-lg-8 col-md-12 col-sm-12">
             {# ACCOUNTS #}
@@ -36,22 +35,6 @@
                     </a>
                 </div>
             </div>
-            {# CATEGORIES #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('categories.index') }}" title="{{ 'categories'|_ }}">{{ 'categories'|_ }}</a></h3>
-
-                </div>
-                <div class="box-body">
-                    <canvas id="categories-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('categories.index') }}" class="btn btn-default button-sm">
-                        <span class="fa fa-bookmark"></span>
-                        <span>{{ 'go_to_categories'|_ }}</span>
-                    </a>
-                </div>
-            </div>
         </div>
         <div class="col-lg-4 col-md-6 col-sm-12">
 
@@ -106,64 +89,6 @@
                     </div>
                 {% endfor %}
             </div>
-
-            {% if billCount > 0 %}
-                {# BILLS #}
-                <div class="box">
-                    <div class="box-header with-border">
-                        <h3 class="box-title"><a href="{{ route('bills.index') }}" title="{{ 'bills'|_ }}">{{ 'bills'|_ }}</a></h3>
-
-                    </div>
-                    <div class="box-body">
-                        <div style="width:100%;margin:0 auto;">
-                            <canvas id="bills-chart" style="width:100%;height:200px;" height="200"></canvas>
-                        </div>
-                    </div>
-                    <div class="box-footer">
-                        <a href="{{ route('bills.index') }}" class="btn btn-default button-sm"><span
-                                    class="fa fa-calendar"></span> {{ 'go_to_bills'|_ }}</a>
-                    </div>
-                </div>
-            {% endif %}
-
-            {# box for piggy bank data (JSON) #}
-            <div id="piggy_bank_overview">
-
-            </div>
-        </div>
-    </div>
-    <div class="row">
-        <div class="col-lg-12 col-md-12 col-sm-12">
-            {# EXPENSE ACCOUNTS #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('accounts.index',['expense']) }}"
-                                             title="{{ 'expense_accounts'|_ }}">{{ 'expense_accounts'|_ }}</a>
-                    </h3>
-                </div>
-                <div class="box-body">
-                    <canvas id="expense-accounts-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('accounts.index', ['expense']) }}" class="btn btn-default button-sm"><span
-                                class="fa fa-shopping-cart"></span> {{ 'go_to_expense_accounts'|_ }}</a>
-                </div>
-            </div>
-            {# OPTIONAL REVENUE ACCOUNTS #}
-            <div class="box">
-                <div class="box-header with-border">
-                    <h3 class="box-title"><a href="{{ route('accounts.index',['revenue']) }}"
-                                             title="{{ 'revenue_accounts'|_ }}">{{ 'revenue_accounts'|_ }}</a></h3>
-
-                </div>
-                <div class="box-body">
-                    <canvas id="revenue-accounts-chart" style="width:100%;height:400px;" height="400" width="100%"></canvas>
-                </div>
-                <div class="box-footer">
-                    <a href="{{ route('accounts.index', ['revenue']) }}" class="btn btn-default button-sm"><span
-                                class="fa fa-download"></span> {{ 'go_to_revenue_accounts'|_ }}</a>
-                </div>
-            </div>
         </div>
     </div>
 {% endblock %}
diff -Naur ../official/resources/views/layout/default.twig ./resources/views/layout/default.twig
--- ../official/resources/views/layout/default.twig	2023-05-28 05:57:50.928996939 +0200
+++ ./resources/views/layout/default.twig	2023-05-28 05:57:52.809020699 +0200
@@ -106,11 +106,22 @@
                         </span>
                     </li>
 
-                    <li class="dropdown user user-menu">
-                    <span style="cursor:default;color:#fff;padding: 15px;display: block;line-height: 20px;">
-                        <span class="hidden-xs">{{ Auth.user.email }}</span>
-                    </span>
+                    <li>
+                        <a href="{{ route('transactions.create', 'withdrawal') }}">
+                            <span class="fa fa-long-arrow-left bg-red" style="font-size:20px;border-radius:50%"></span>
+                        </a>
                     </li>
+                    <li>
+                        <a href="{{ route('transactions.create', 'deposit') }}">
+                            <span class="fa fa-long-arrow-right bg-green" style="font-size:20px;border-radius:50%"></span>
+                        </a>
+                    </li>
+                    <li>
+                        <a href="{{ route('transactions.create', 'transfer') }}">
+                            <span class="fa fa-exchange bg-blue" style="font-size:20px;border-radius:50%"></span>
+                        </a>
+                    </li>
+
                     <li id="sidebar-toggle">
                         <a href="#" data-toggle="control-sidebar"><span class="fa fa-plus-circle"></span></a>
                     </li>
diff -Naur ../official/resources/views/list/groups-tiny.twig ./resources/views/list/groups-tiny.twig
--- ../official/resources/views/list/groups-tiny.twig	2023-04-15 11:46:01.085659939 +0200
+++ ./resources/views/list/groups-tiny.twig	2023-04-15 17:32:17.998943840 +0200
@@ -7,7 +7,12 @@
             {{ transaction.description }}
 
             <span class="pull-right small">
-          {% if transaction.transaction_type_type == 'Deposit' %}
+          {% if transaction.date > date('now') %}
+              {#  future transactions:  #}
+              <span class="text-secondary">
+                  {{ formatAmountBySymbol(transaction.amount|abs, transaction.currency_symbol, transaction.currency_decimal_places, false) }}
+              </span>
+          {% elseif transaction.transaction_type_type == 'Deposit' %}
               {{ formatAmountBySymbol(transaction.amount*-1, transaction.currency_symbol, transaction.currency_decimal_places) }}
               {% if null != transaction.foreign_amount %}
                   ({{ formatAmountBySymbol(transaction.foreign_amount*-1, transaction.foreign_currency_symbol, transaction.foreign_currency_decimal_places) }})
diff -Naur ../official/resources/views/preferences/index.twig ./resources/views/preferences/index.twig
--- ../official/resources/views/preferences/index.twig	2023-04-15 11:46:01.089659964 +0200
+++ ./resources/views/preferences/index.twig	2023-04-15 17:32:17.998943840 +0200
@@ -196,30 +196,24 @@
                                         <div class="radio">
                                             <label>
                                                 <input type="radio" name="viewRange"
-                                                       value="last7" {% if viewRange == 'last7' %} checked {% endif %}>
-                                                {{ 'pref_last7'|_ }}
+                                                       value="3W" {% if viewRange == '3W' %} checked {% endif %}>
+                                                {{ 'pref_3W'|_ }}
                                             </label>
                                         </div>
 
                                         <div class="radio">
                                             <label>
                                                 <input type="radio" name="viewRange"
-                                                       value="1M" {% if viewRange == '1M' %} checked {% endif %}>
-                                                {{ 'pref_1M'|_ }}
-                                            </label>
-                                        </div>
-                                        <div class="radio">
-                                            <label>
-                                                <input type="radio" name="viewRange"
-                                                       value="last30" {% if viewRange == 'last30' %} checked {% endif %}>
-                                                {{ 'pref_last30'|_ }}
+                                                       value="last7" {% if viewRange == 'last7' %} checked {% endif %}>
+                                                {{ 'pref_last7'|_ }}
                                             </label>
                                         </div>
+
                                         <div class="radio">
                                             <label>
                                                 <input type="radio" name="viewRange"
-                                                       value="MTD" {% if viewRange == 'MTD' %} checked {% endif %}>
-                                                {{ 'pref_MTD'|_ }}
+                                                       value="1M" {% if viewRange == '1M' %} checked {% endif %}>
+                                                {{ 'pref_1M'|_ }}
                                             </label>
                                         </div>
 
diff -Naur ../official/app/Http/Controllers/Account/ReconcileController.php ./app/Http/Controllers/Account/ReconcileController.php
--- ../official/app/Http/Controllers/Account/ReconcileController.php	2023-07-01 10:55:37.274919027 +0200
+++ ./app/Http/Controllers/Account/ReconcileController.php	2023-07-01 10:55:39.135019911 +0200
@@ -99,18 +99,15 @@
         }
         $currency = $this->accountRepos->getAccountCurrency($account) ?? app('amount')->getDefaultCurrency();
 
-        // no start or end:
-        $range = app('navigation')->getViewRange(false);
-
-        // get start and end
-
+        // compute values of 'start' and 'end' whatever the preferred range is
         if (null === $start && null === $end) {
-            /** @var Carbon $start */
-            $start = clone session('start', app('navigation')->startOfPeriod(new Carbon(), $range));
-            /** @var Carbon $end */
-            $end = clone session('end', app('navigation')->endOfPeriod(new Carbon(), $range));
+           /** @var Carbon $start */
+           $start = clone (new Carbon)->subMonth();
+           /** @var Carbon $end */
+           $end = clone (new Carbon);
         }
         if (null === $end) {
+            $range = app('preferences')->get('viewRange', '1M')->data;
             /** @var Carbon $end */
             $end = app('navigation')->endOfPeriod($start, $range);
         }
diff -Naur ../official/app/Http/Controllers/Chart/AccountController.php ./app/Http/Controllers/Chart/AccountController.php
--- ../official/app/Http/Controllers/Chart/AccountController.php	2023-07-01 10:55:37.278919254 +0200
+++ ./app/Http/Controllers/Chart/AccountController.php	2023-07-01 10:55:39.139020110 +0200
@@ -495,7 +495,7 @@
                 $current->addDay();
             }
         }
-        if ('1W' === $step || '1M' === $step || '1Y' === $step) {
+        if ('1W' === $step || '3W' === $step || '1M' === $step || '1Y' === $step) {
             while ($end >= $current) {
                 $balance         = (float)app('steam')->balance($account, $current, $currency);
                 $label           = app('navigation')->periodShow($current, $step);
diff -Naur ../official/app/Jobs/CreateRecurringTransactions.php ./app/Jobs/CreateRecurringTransactions.php
--- ../official/app/Jobs/CreateRecurringTransactions.php	2023-07-01 10:55:37.282919480 +0200
+++ ./app/Jobs/CreateRecurringTransactions.php	2023-07-01 10:55:39.147020508 +0200
@@ -92,6 +92,8 @@
         $this->created           = 0;
         $this->recurrences       = new Collection();
         $this->groups            = new Collection();
+        $this->days_max_late     = 8;
+        $this->days_max_early    = 8;
 
         Log::debug(sprintf('Created new CreateRecurringTransactions("%s")', $this->date->format('Y-m-d')));
     }
@@ -332,10 +334,10 @@
             );
 
             // start looping from $startDate to today perhaps we have a hit?
-            // add two days to $this->date, so we always include the weekend.
-            $includeWeekend = clone $this->date;
-            $includeWeekend->addDays(2);
-            $occurrences = $this->repository->getOccurrencesInRange($repetition, $recurrence->first_date, $includeWeekend);
+            // add few days to anticipate next occurrences
+            $endDate = clone $this->date;
+            $endDate->addDays($this->days_max_early);
+            $occurrences = $this->repository->getOccurrencesInRange($repetition, $recurrence->first_date, $endDate);
 
             unset($includeWeekend);
 
@@ -383,10 +385,18 @@
     private function handleOccurrence(Recurrence $recurrence, RecurrenceRepetition $repetition, Carbon $date): ?TransactionGroup
     {
         $date->startOfDay();
-        if ($date->ne($this->date)) {
+
+        // allow some margin in the future (to anticipate) and in the past (in case cron was not executed)
+        $datediff = $date->diffInDays($this->date);
+        if ($datediff > $this->days_max_early) {
+            //print "date too much in the future\n";
+            return null;
+        }
+        if ($datediff < -$this->days_max_late) {
+            //print "date too much in the past\n";
             return null;
         }
-        Log::debug(sprintf('%s IS today (%s)', $date->format('Y-m-d'), $this->date->format('Y-m-d')));
+        Log::debug(sprintf('%s IS close to today (%s)', $date->format('Y-m-d'), $this->date->format('Y-m-d')));
 
         // count created journals on THIS day.
         $journalCount = $this->repository->getJournalCount($recurrence, $date, $date);
diff -Naur ../official/app/Support/Http/Controllers/GetConfigurationData.php ./app/Support/Http/Controllers/GetConfigurationData.php
--- ../official/app/Support/Http/Controllers/GetConfigurationData.php	2023-07-01 10:55:37.302920613 +0200
+++ ./app/Support/Http/Controllers/GetConfigurationData.php	2023-07-01 10:55:39.163021305 +0200
@@ -133,10 +133,15 @@
         }
 
         // today:
-        /** @var Carbon $todayStart */
-        $todayStart = app('navigation')->startOfPeriod($today, $viewRange);
-        /** @var Carbon $todayEnd */
-        $todayEnd = app('navigation')->endOfPeriod($todayStart, $viewRange);
+        if ('3W' === $viewRange) {
+            $todayStart = Carbon::now()->subDays(14);
+            $todayEnd = Carbon::now()->addDays(7);
+        } else {
+            /** @var Carbon $todayStart */
+            $todayStart = app('navigation')->startOfPeriod($today, $viewRange);
+            /** @var Carbon $todayEnd */
+            $todayEnd = app('navigation')->endOfPeriod($todayStart, $viewRange);
+        }
         if ($todayStart->ne($start) || $todayEnd->ne($end)) {
             $ranges[ucfirst((string)trans('firefly.today'))] = [$todayStart, $todayEnd];
         }
diff -Naur ../official/app/Support/Navigation.php ./app/Support/Navigation.php
--- ../official/app/Support/Navigation.php	2023-07-01 10:55:37.302920613 +0200
+++ ./app/Support/Navigation.php	2023-07-01 10:55:39.163021305 +0200
@@ -53,6 +53,7 @@
             '1W'        => 'addWeeks',
             'weekly'    => 'addWeeks',
             'week'      => 'addWeeks',
+            '3W'        => 'addWeeks',
             '1M'        => 'addMonths',
             'month'     => 'addMonths',
             'monthly'   => 'addMonths',
@@ -75,6 +76,7 @@
             'YTD'       => 'addYears',
         ];
         $modifierMap = [
+            '3W'        => 3,
             'quarter'   => 3,
             '3M'        => 3,
             'quarterly' => 3,
@@ -230,7 +232,7 @@
         }
 
 
-        if ('custom' === $repeatFreq) {
+        if ('custom' === $repeatFreq || '3W' === $repeatFreq) {
             return $date; // the date is already at the start.
         }
         Log::error(sprintf('Cannot do startOfPeriod for $repeat_freq "%s"', $repeatFreq));
@@ -254,6 +256,7 @@
             '1W'        => 'addWeek',
             'week'      => 'addWeek',
             'weekly'    => 'addWeek',
+            '3W'        => 'addWeeks',
             '1M'        => 'addMonth',
             'month'     => 'addMonth',
             'monthly'   => 'addMonth',
@@ -268,6 +271,7 @@
             '1Y'        => 'addYear',
         ];
         $modifierMap = [
+            '3W'        => 3,
             'quarter'   => 3,
             '3M'        => 3,
             'quarterly' => 3,
@@ -276,7 +280,7 @@
             '6M'        => 6,
         ];
 
-        $subDay = ['week', 'weekly', '1W', 'month', 'monthly', '1M', '3M', 'quarter', 'quarterly', '6M', 'half-year', 'half_year', '1Y', 'year', 'yearly'];
+        $subDay = ['week', 'weekly', '1W', 'month', 'monthly', '3W', '1M', '3M', 'quarter', 'quarterly', '6M', 'half-year', 'half_year', '1Y', 'year', 'yearly'];
 
         // if the range is custom, the end of the period
         // is another X days (x is the difference between start)
@@ -645,6 +649,10 @@
 
             return $date;
         }
+        if ('3W' === $repeatFreq) {
+            $date->subDays(21);
+            return $date;
+        }
         // a custom range requires the session start
         // and session end to calculate the difference in days.
         // this is then subtracted from $theDate (* $subtract).
@@ -714,6 +722,11 @@
 
             return $end;
         }
+
+        if ('3W' === $range) {
+            $end->addDays(21);
+            return $end;
+        }
         if ('6M' === $range) {
             if ($start->month >= 7) {
                 $end->endOfYear();
@@ -799,6 +812,9 @@
             case 'last7':
                 $start->subDays(7);
                 return $start;
+            case '3W':
+                $start->subDays(14);
+                return $start;
             case 'last30':
                 $start->subDays(30);
                 return $start;
